packages/core/src/browser/common-frontend-contribution.ts

```diff
@@ -638,40 +643,45 @@ export class CommonFrontendContribution implements FrontendApplicationContributi
         });
     }
 
-    private findTabBar(event?: Event): TabBar<Widget> | undefined {
-        if (event && event.target) {
-            const tabBar = this.shell.findWidgetForElement(event.target as HTMLElement);
-            if (tabBar instanceof TabBar) {
-                return tabBar;
-            }
+    /**
+     * Evaluates the currentIndex of the title in the array of titles.
+     * @param event: `event` to be used when searching for the title and the tab-bar.
+     *
+     * @returns `currentIndex` if the `targetTitle` is available in the array, else returns the index of currently-selected title.
+     */
+    private targetTitleIndex(event?: Event): number {
+        const tabBar = this.contextMenuService.findTabBar(event)!;
+        const targetTitle = this.contextMenuService.findTitle(tabBar, event);
+        let currentIndex: number;
+        if (targetTitle) {
+            currentIndex = tabBar.titles.indexOf(targetTitle);
+        } else {
+            currentIndex = tabBar.currentIndex;
         }
-        return this.shell.currentTabBar;
+        return currentIndex;
+    }
+
+    private canToggleMaximized(event?: Event): boolean {
+        const targetTabBar = this.contextMenuService.findTabBar(event);
+        if (targetTabBar) {
+            return this.shell.canToggleMaximized({ targetTabBar });
```

Comment:
Why so complicated? You can pass down `event` and if it is defined just check whether target belongs to node of main or bottom areas by `node.contains`. Otherwise fall back to old logic.